parser.js

```diff
@@ -762,7 +762,7 @@
         //      PatternCharacter
         return createCharacter(res);
       }
-      else if (!isUnicodeMode && (res = matchReg(/^(?:]|})/))) {
+      else if (!isUnicodeMode && (res = matchReg(/^(?:\]|\})/))) {
```

Comment:
Aside: Should we introduce an `annexB: boolean` option? When this option is `false`, it will reject any annexB productions. We can default it to `true` to avoid breaking changes, but users can opt-in to the modern stricter regex syntax.

Reply:
More than happy to accept a PR implementing this!